### PR TITLE
RTCPeerConnection.addIceCandidate() - remove RTCIceCandidateInit dict…

### PR DESCRIPTION
**Pull request by hamishwillee**
[RTCPeerConnection.addIceCandidate()](https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/addIceCandidate) takes a dictionary option [RTCIceCandidateInit](https://developer.mozilla.org/en-US/docs/Web/API/RTCIceCandidateInit). This PR takes all of the relevant info from `RTCIceCandidateInit` and the `RTCIceCandidateInit` properties and puts it into an options argument in `RTCPeerConnection.addIceCandidate()`.

I have also linked each property to the matching properties in [RTCIceCandidate](https://developer.mozilla.org/en-US/docs/Web/API/RTCIceCandidate) (e.g. [RTCIceCandidate.sdpMid](https://developer.mozilla.org/en-US/docs/Web/API/RTCIceCandidate/sdpMid) as these sometimes have additional information/examples).

This is first step in the activity. IFF this looks OK I will:
- Propagate the same changes to other things that use the `RTCIceCandidateInit`, including 
  - [`RTCIceCandidate` constructor](https://developer.mozilla.org/en-US/docs/Web/API/RTCIceCandidate/RTCIceCandidate) 
  - [`RTCIceCandidate.toJSON()` method return value ](https://developer.mozilla.org/en-US/docs/Web/API/RTCIceCandidate/toJSON)
- Delete the  `RTCIceCandidateInit` page and it's property pages
- Delete/tidy up any mention of `RTCIceCandidateInit` 

I'm also looking at BCD work for this in https://github.com/mdn/browser-compat-data/pull/11941

### CODE REVIEW

File: files/en-us/web/api/rtcicecandidate/sdpmid/index.html [hunk 1]
```diff
@@ -44,10 +44,8 @@ <h3 id="Value">Value</h3>
   the candidate.</p>
 
 <div class="note">
-  <p><strong>Note:</strong> Attempting to add a candidate (using
-    {{domxref("RTCPeerConnection.addIceCandidate", "addIceCandidate()")}}) that has a
-    value of <code>null</code> for either <code>sdpMid</code> or
-    <code>sdpMLineIndex</code> will throw a <code>TypeError</code> exception.</p>
+  <p><strong>Note:</strong> Attempting to add a candidate (using {{domxref("RTCPeerConnection.addIceCandidate", "addIceCandidate()")}}) that has a
```
Comment (hamishwillee) [inline files/en-us/web/api/rtcicecandidate/sdpmid/index.html:47]: This was just wrong - it's an error if both values are null, not if "either"

File: files/en-us/web/api/rtcpeerconnection/addicecandidate/index.html [hunk 2]
```diff
@@ -17,65 +17,110 @@
...
-		<p>If no <code>candidate</code> object is specified, or its value is
-			<code>null</code>, an end-of-candidates signal is sent to the remote peer
-			using the <code>end-of-candidates</code> a-line, formatted like this:</p>
-
+		<p>An {{domxref("RTCIceCandidate")}} object, or an object that has the following properties:</p>
+		<!-- The spec calls this object an RTCIceCandidateInit -->
+			<dl>
+				<dt><code>candidate</code> {{optional_inline}}</dt>
+				<dd><p>A {{domxref("DOMString")}} describing the properties of the candidate, taken directly
+						from the {{Glossary("SDP")}} attribute <code>"candidate"</code>. The candidate string
```
Comment (teoli2003) [inline files/en-us/web/api/rtcpeerconnection/addicecandidate/index.html:48]: The .´{{Glossary("SDP")}}´ doesn't exist. But we can link to https://developer.mozilla.org/en-US/docs/Web/API/WebRTC_API/Protocols#sdp instead.

**Review by teoli2003 — CHANGES_REQUESTED** [review]
One red link to fix.

File: files/en-us/web/api/rtcpeerconnection/addicecandidate/index.html (hunk 2 above)
Comment (hamishwillee) [reply files/en-us/web/api/rtcpeerconnection/addicecandidate/index.html]: Thanks very much @teoli2003 ! Link fixed.

**Review by teoli2003 — APPROVED** [review]

**Pull request merged**